ui/app/components/ProcSelectComponent/ProcSelectComponent.js

```diff
@@ -7,67 +7,94 @@ import safeTraverse from 'utils/safeTraverse';
-    if (didAppChange || didClusterChange) {
-      this.props.getProcs({
-        app: nextProps.app,
-        cluster: nextProps.cluster,
-      });
+    if (didAppChange || didClusterChange || nextProps.isPolicyPage !== this.props.isPolicyPage) {
+      if (nextProps.isPolicyPage) {
+        this.props.getPolicyProcs(nextProps.app)(nextProps.cluster)();
+      } else {
+        this.props.getProcs(nextProps.app)(nextProps.cluster)();
+      }
     }
   }
 
-  render () {
-    const { onChange, procs } = this.props;
-    const procList = procs.asyncStatus === 'SUCCESS'
-      ? procs.data.map(c => ({ name: c })) : [];
-    const valueOption = this.props.value && { name: this.props.value };
+  render() {
+    const {onChange, procs, policyProcs} = this.props;
+    const finalProcs = this.props.isPolicyPage ? policyProcs : procs;
+    const procList = finalProcs.asyncStatus === 'SUCCESS'
+      ? finalProcs.data.map(c => ({name: c})) : [];
+    const valueOption = this.props.value && {name: this.props.value};
     return (
       <div>
         <label className={styles.label} htmlFor="proc">Process</label>
+        {this.props.isPolicyPage &&
+        <Creatable
+          id="proc"
+          clearable={false}
+          options={procList}
+          onChange={onChange || noop}
+          labelKey="name"
+          valueKey="name"
+          onInputChange={debounce(this.props.getPolicyProcs(this.props.app)(this.props.cluster), 500)}
+          isLoading={finalProcs.asyncStatus === 'PENDING'}
+          value={valueOption}
+          noResultsText={finalProcs.asyncStatus !== 'PENDING' ? 'No results found!' : 'Searching...'}
+          placeholder="Type to search..."
+          promptTextCreator={(label) => "Add process: " + label}
+        />}
+        {!this.props.isPolicyPage &&
         <Select
           id="proc"
           clearable={false}
           options={procList}
           onChange={onChange || noop}
           labelKey="name"
           valueKey="name"
-          isLoading={procs.asyncStatus === 'PENDING'}
+          onInputChange={debounce(this.props.getProcs(this.props.app)(this.props.cluster), 500)}
+          isLoading={finalProcs.asyncStatus === 'PENDING'}
           value={valueOption}
-          noResultsText={procs.asyncStatus !== 'PENDING' ? 'No results found!' : 'Searching...'}
+          noResultsText={finalProcs.asyncStatus !== 'PENDING' ? 'No results found!' : 'Searching...'}
           placeholder="Type to search..."
-        />
+        />}
       </div>
     );
   }
 }
 
 const mapStateToProps = (state, ownProps) => ({
   procs: safeTraverse(state, ['procs', ownProps.cluster]) || {},
+  policyProcs: safeTraverse(state, ['policyProcs', ownProps.cluster]) || {},
 });
 
 const mapDispatchToProps = dispatch => ({
-  getProcs: params => dispatch(fetchProcsAction(params)),
+  getProcs: app => cluster => event => dispatch(fetchProcsAction(app, cluster, event)),
```

Comment:
fully qualify all the old getters for app, cluster, proc.
getProfileProcs

Reply:
Made all getter names fully qualified.